pkg/workflow/compiler.go

```diff
@@ -3125,6 +3125,33 @@ func generatePlaywrightDockerArgs(playwrightTool any, networkPermissions *Networ
 	}
 }
 
+// hasContentsPermission checks if the given permissions string includes contents access
+// Returns true if permissions include contents: read, contents: write, read-all, write-all, read, or write
+func (c *Compiler) hasContentsPermission(permissions string) bool {
+	if permissions == "" {
+		return false
+	}
+
+	// Convert to lowercase for case-insensitive matching
+	permsLower := strings.ToLower(permissions)
```

Comment:
Feels like we need to YAML parse the permissions rather than just treating it as a string - e.g. what if there are comments etc.

Reply:
@dsyme the way we treat permissions is horrible. we should parse it into a structure object, and have decent apis to mutate it.

Reply:
but i did not want this to snowball.